test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -795,6 +808,8 @@ protected void doClose() throws IOException {
             }
         } catch (InterruptedException e) {
             throw new IllegalStateException(e);
+        } finally {
+            ThreadPool.terminate(testExecutor, 10, TimeUnit.SECONDS);
```

Comment:
Could we have an `assertTrue` here to make sure it did actually terminate?

Reply:
Yep see 63a461498b049ba8ae0b39b924b9bc514f0484da